internal/services/cosmos/cosmosdb_postgresql_cluster_resource.go

```diff
@@ -0,0 +1,631 @@
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			model := resp.Model
+			if model == nil {
+				return fmt.Errorf("retrieving %s: model was nil", id)
+			}
+
+			state := CosmosDbPostgreSQLClusterModel{
+				Name:              id.ServerGroupsv2Name,
+				ResourceGroupName: id.ResourceGroupName,
+				Location:          location.Normalize(model.Location),
+			}
+
+			if props := model.Properties; props != nil {
+				state.AdministratorLoginPassword = metadata.ResourceData.Get("administrator_login_password").(string)
+				state.SourceResourceId = metadata.ResourceData.Get("source_resource_id").(string)
+				state.SourceLocation = metadata.ResourceData.Get("source_location").(string)
+				state.PointInTimeInUTC = metadata.ResourceData.Get("point_in_time_in_utc").(string)
+				state.CoordinatorPublicIPAccessEnabled = *props.CoordinatorEnablePublicIPAccess
+				state.CoordinatorServerEdition = *props.CoordinatorServerEdition
+				state.CoordinatorStorageQuotaInMb = *props.CoordinatorStorageQuotaInMb
+				state.CoordinatorVCoreCount = *props.CoordinatorVCores
+				state.HaEnabled = *props.EnableHa
+				state.NodeCount = *props.NodeCount
+				state.NodePublicIPAccessEnabled = *props.NodeEnablePublicIPAccess
+				state.NodeServerEdition = *props.NodeServerEdition
+				state.NodeStorageQuotaInMb = *props.NodeStorageQuotaInMb
+				state.NodeVCores = *props.NodeVCores
+				state.ShardsOnCoordinatorEnabled = *props.EnableShardsOnCoordinator
+
+				if v := props.CitusVersion; v != nil {
+					state.CitusVersion = *v
+				}
+
+				if v := props.MaintenanceWindow; v != nil {
+					state.MaintenanceWindow = flattenMaintenanceWindow(v)
+				}
+
+				if v := props.PreferredPrimaryZone; v != nil {
+					state.PreferredPrimaryZone = *v
+				}
+
+				if v := props.PostgresqlVersion; v != nil {
+					state.SqlVersion = *v
+				}
+
+				if v := props.EarliestRestoreTime; v != nil {
+					state.EarliestRestoreTime = *v
+				}
+			}
+
+			if model.Tags != nil {
+				state.Tags = *model.Tags
+			}
+
+			return metadata.Encode(&state)
+		},
+	}
+}
+
+func (r CosmosDbPostgreSQLClusterResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 3 * time.Hour,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Cosmos.ClustersClient
+
+			id, err := clusters.ParseServerGroupsv2ID(metadata.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			if err := client.DeleteThenPoll(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %+v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func expandMaintenanceWindow(input []MaintenanceWindow) *clusters.MaintenanceWindow {
+	if len(input) == 0 {
+		return &clusters.MaintenanceWindow{
+			CustomWindow: utils.String("Disabled"),
+		}
+	}
+
+	v := input[0]
+
+	maintenanceWindow := clusters.MaintenanceWindow{
+		CustomWindow: utils.String("Enabled"),
```

Comment:
These can be swapped out for `pointer.To`
```suggestion
		CustomWindow: pointer.To("Enabled"),
```

Reply:
Updated